src/macros.php

```diff
@@ -297,3 +299,49 @@
         return $this->reverse()->after($currentItem, $fallback);
     });
 }
+
+if (! Collection::hasMacro('paginate')) {
+    /*
+     * Paginate the given collection
+     *
+     * @param int $perPage
+     * @param int $total
+     * @param int $page
+     * @param string $pageName
+     *
+     * @return \Illuminate\Pagination\LengthAwarePaginator
+     */
+    Collection::macro('paginate', function (int $perPage = 15, string $pageName = 'page', int $page = null, int $total = null, array $options = []): LengthAwarePaginator {
```

Comment:
Didn't realize the `Builder` used that parameter order. In that case, ignore my inital commend!

I also agree that columns aren't useful here. If people really need to selectively paginate data, they can `map` their collection first.